src/firstvoices/backend/models/characters.py

```diff
@@ -143,21 +146,43 @@ def __str__(self):
         return self.title
 
 
-# See fw-4172: AlphabetMapper model implementation once g2p work has been completed
-# class AlphabetMapper(BaseSiteContentModel):
-#     """
-#     Represents a private table holding the g2p mappings and configuration for a site.
-#     """
-#
-#     class Meta:
-#         verbose_name = _("alphabet mapper")
-#         verbose_name_plural = _("alphabet mappers")
-#
-#     # See fw-4172: Once file storage is configured, enable these FileFields / change to proper paths
-#     # alphabet_mapper is a placeholder for now.
-#     # input_to_canonical = models.FileField(upload_to="alphabet_mapper")
-#     # canonical_to_base = models.FileField(upload_to="alphabet_mapper")
-#     # g2p_config = models.FileField(upload_to="alphabet_mapper")
-#
-#     def __str__(self):
-#         return self.id
+class ConfusableMapper(BaseSiteContentModel):
+    """
+    Represents a private table holding the confusable g2p mapping and configuration for a site.
+    """
+
+    class Meta:
+        verbose_name = _("confusable mapper")
+        verbose_name_plural = _("confusable mappers")
+
+    # from fv-alphabet:confusable_characters
+    # JSON representation of a g2p mapping from confusable characters to canonical characters
+    input_to_canonical_map = models.JSONField()
+    # TODO: Possibly write a custom field that takes YAML here
+    g2p_config_yaml = models.TextField(blank=True)
```

Comment:
You could store this as JSON internally and convert it to yaml when needed using the standard json and yaml modules: https://www.pythonforbeginners.com/basics/convert-json-to-yaml-in-python